app/views/products/_product.html.erb

```diff
@@ -1,6 +1,14 @@
 <tr id="<%= dom_id product %>" class="<%= cycle('list_line_odd', 'list_line_even') %>">
   <td class="description">
     <h1><%= product.title %></h1>
+    <div>Users Ratings: <%= average_rating(product) %></div>
+
+    <%= form_with url: ratings_path, method: :post, local: false do |form| %>
+      <%= form.hidden_field :product_id, value: product.id %>
+      <%= form.select :rating, generate_ratings, selected: current_product_rating_by_user(product.ratings, @logged_in_user) %>
```

Comment:
what if user is not logged in?

Reply:
/books route which lists all products is not accessible if user is not logged in